client/src/templates/layouts/NavigationLayout.js

```diff
@@ -111,11 +139,236 @@ const NavigationLayout = (props) => {
   const history = useHistory();
 
   const [drawerOpened, setDrawerOpened] = useState(false);
+  const [changeValue, setChangeValue] = useState(false);
+  const [feedbackState, feedbackDispatch] = useReducer(
+    feedbackReducer,
+    initialState.feedbackReducer,
+  );
+
+  const [feedbackFormState, feedbackFormDispatch] = useReducer(
+    feedbackFormReducer,
+    initialState.feedbackFormReducer,
+  );
+
+  const {
+    ratingModal,
+    textFeedbackModal,
+    radioModal,
+    thanksModal,
+    rating,
+    mostValuableFeature,
+    whatWouldChange,
+    generalFeedback,
+    age,
+    covidImpact,
+  } = feedbackState;
+
+  const dispatchAction = (type, key, value) => {
+    feedbackDispatch({ type, key, value });
+  };
 
   const toggleDrawer = () => {
     setDrawerOpened(!drawerOpened);
   };
 
+  const toggleModal = (modalName) => {
+    dispatchAction(TOGGLE_STATE, modalName);
+  };
+
+  const closeRatingModal = (ratingValue) => {
+    if (drawerOpened) {
+      toggleDrawer();
+    }
+    dispatchAction(SET_VALUE, "rating", ratingValue);
+    toggleModal("ratingModal");
+    toggleModal("textFeedbackModal");
+  };
+
+  const closeTextFeedbackModal = () => {
+    toggleModal("textFeedbackModal");
+    toggleModal("radioModal");
+  };
+
+  const closeRadioModal = () => {
+    submitFeedbackForm();
+    toggleModal("thanksModal");
+    toggleModal("radioModal")
+    if (feedbackFormState.error === "") {
```

Comment:
You are not doing anything with this check here. You need to put the `toggleModal("thanksModal")` inside of here for it to have the desired effect.

Reply:
I just throw it right here because if you wrap it around in the `thanks modal` then the `thanksModal` wont work. Im not sure why.

Reply:
It won't work because you have already sent feedback from your ip and therefore get an error. That is the way it should work.